### PR TITLE
Validate installer arguments

### PR DESCRIPTION
**Pull request by lferran**
This PR aims to fix https://github.com/canonical/microk8s/issues/2915

Essentially, it adds a some validations on windows installer cli arguments.

### CODE REVIEW

File: installer/cli/microk8s.py [hunk 1]
```diff
@@ -104,55 +104,62 @@ def show_help():
...
-        aux = Linux(args)
-        if not aux.is_enough_space():
-            echo.warning("VM disk size requested exceeds free space on host.")
+    host = {"win32": Windows, "darwin": MacOS, "linux": Linux}[platform](args)
+    if not host.has_enough_disk_space():
+        echo.warning("VM disk size requested exceeds free space on host.")
+    if not host.has_enough_cpus():
+        echo.warning("VM cpus requested exceed number of available cores on host.")
+    if not host.has_enough_memory():
+        echo.warning("VM memory requested exceeds the total memory on host.")
```
Comment (lferran) [inline installer/cli/microk8s.py:162]: Is it OK to just emit a warning in these cases?

File: .github/workflows/build-installer.yml [hunk 2]
```diff
@@ -25,7 +25,7 @@ jobs:
         run: python -m pip install -r ../requirements.txt
       - name: Build exe
         working-directory: ${{ github.workspace }}/installer
-        run: pyinstaller.exe --onefile ./microk8s.spec
+        run: pyinstaller.exe ./microk8s.spec
```
Comment (lferran) [inline .github/workflows/build-installer.yml:28]: I fixed this after I saw [this error in GH actions](https://github.com/canonical/microk8s/runs/6386624112?check_suite_focus=true#step:5:12)

File: installer/cli/microk8s.py (hunk 1 above)
Comment (ktsakalozos) [reply installer/cli/microk8s.py:178]: I think we need to not proceed with the wrong CPU size. Here is what I got:
```
jackal@aurora:~/workspace/microk8s/installer$ ./microk8s install --cpu 1000
VM cpus requested exceed number of available cores on host.
launch failed: The following errors occurred:                                   
qemu-system-x86_64: Invalid SMP CPUs 1000. The max CPUs supported by machine 'pc-i440fx-focal' is 255

launch failed: instance "microk8s-vm" already exists                            


launch failed: instance "microk8s-vm" already exists                            


launch failed: instance "microk8s-vm" already exists                            


An error occurred with the instance when trying to launch with 'multipass': returned exit code 2.
Ensure that 'multipass' is setup correctly and try again.
``` 

Resulting to a failed installation.

Comment (ktsakalozos) [reply installer/cli/microk8s.py:178]: Same with more VM memory:
```
jackal@aurora:~/workspace/microk8s/installer$ ./microk8s install --mem 64 
VM memory requested exceeds the total memory on host.
Launched: microk8s-vm                                                           
An error occurred when trying to execute 'sudo ping -c 1 snapcraft.io' with 'multipass': returned exit code 1.
```

**Review by ktsakalozos — APPROVED** [review]
LGTM +1 thank you @lferran

**Pull request merged**